lib/console.js

```diff
@@ -1,30 +1,41 @@
-'use babel';
+"use babel";
 
-const CONSOLE = {};
+export default class Console {
+  static get enabled() {
+    if (process.env.SILENCE_LOG) {
+      return false;
+    }
 
-function getEnabled () {
-  if (process.env.SILENCE_LOG) { return false; }
-  return atom.config.get('linter-eslint-node.advanced.enableLogging');
-}
+    return atom.config.get("linter-eslint-node.advanced.enableLogging");
+  }
 
-let isEnabled = getEnabled();
+  static log(...args) {
+    if (this.enabled) {
+      window.console.log("[linter-eslint-node]", ...args);
+    }
+  }
 
-atom.config.observe('linter-eslint-node.advanced.enableLogging', () => {
-  isEnabled = getEnabled();
-});
+  static debug(...args) {
+    if (this.enabled) {
+      window.console.debug("[linter-eslint-node]", ...args);
+    }
+  }
 
-function makeConsoleMethod (name) {
-  return (...args) => {
-    if (!isEnabled) { return; }
-    return window.console[name](
-      '[linter-eslint-node]',
-      ...args
-    );
-  };
-}
+  static info(...args) {
+    if (this.enabled) {
+      window.console.info("[linter-eslint-node]", ...args);
+    }
+  }
 
-['log', 'warn', 'error', 'info', 'debug', 'group'].forEach(name => {
-  CONSOLE[name] = makeConsoleMethod(name);
-});
```

Comment:
Some of this really feels like a rewrite just for rewrite's sake.

Reply:
In this case it was just about editor hits 😅 
<img width="323" alt="image" src="https://github.com/pulsar-linter/linter-eslint-node/assets/2230835/9b09834e-80a1-4afb-9e58-12a83409e7e5">
<img width="802" alt="image" src="https://github.com/pulsar-linter/linter-eslint-node/assets/2230835/4819c222-f247-42b9-ab55-0a5e539349ec">

Reply:
🤔 Thinking about this, something like this may be cleaner and produce better type hints:
🤷‍♂️ Imma leave it for now

```js
'use babel';

function isEnabled() {
  if (process.env.SILENCE_LOG) {
    return false;
  }

  return atom.config.get('linter-eslint-node.advanced.enableLogging');
}

export default {
  log(...args) {
    if (isEnabled()) {
      window.console.log('[linter-eslint-node]', ...args);
    }
  },
  debug(...args) {
    if (isEnabled()) {
      window.console.debug('[linter-eslint-node]', ...args);
    }
  },
  info(...args) {
    if (isEnabled()) {
      window.console.info('[linter-eslint-node]', ...args);
    }
  },
  warn(...args) {
    if (isEnabled()) {
      window.console.warn('[linter-eslint-node]', ...args);
    }
  },
  error(...args) {
    if (isEnabled()) {
      window.console.error('[linter-eslint-node]', ...args);
    }
  }
};
```